CLI.md

```diff
@@ -0,0 +1,105 @@
+the [readme](README.md). Basic use of the CLI such as [creating a rule](CONTRIBUTING.md#creating-a-rule-with-the-cli) or 
+[testing](CONTRIBUTING.md#testing-a-rule-with-the-cli) are referenced in the [contribution guide](CONTRIBUTING.md).
+
+
+## Using a config file or environment variables
+
+CLI commands which are tied to Kibana and Elasticsearch are capable of parsing auth-related keyword args from a config 
+file or environment variables. Any command option which calls `callback=set_param_values` will search for set kwargs.
+
+If a value is set in multiple places, such as config file and environment variable, the order of precedence will be as 
+follows:
+* explicitly passed args (such as `--user joe`)
+* environment variables
+* config values
+* prompt (this only applies to certain values)
+
+#### Setup a config file
+
+In the root directory of this repo, create the file `.detection-rules-cfg.json` and add relevant values
+
+#### Using environment variables
+
+Environment variables using the key format: `DR_<UPPERCASED_KEY_NAME>` will be parsed in commands which expect it.
+EX: `DR_USER=joe`
+
+## Importing rules into the repo
+
+You can import rules into the repo using the `create-rule` or `import-rules` commands. Both of these commands will 
+require that the rules are schema-compliant and able to pass full validation. The biggest benefit to using these 
+commands is that they will strip[*](#i18n-note) additional fields[**](#i18n-note-2) and prompt for missing required 
+fields. 
+
+Alternatively, you can manually place rule files in the directory and run tests to validate as well.
+
+<a id="i18n-note">\* Note</a>: This is currently limited to flat fields and may not apply to nested values.<br>
+<a id="i18n-note-2">\** Note</a>: Additional fields are based on the current schema at the time the command is used.
+
+
+#### `create-rule`
+
+This command will allow you to pass a rule file using the `-c/--config` parameter. This is limited to one rule at a time
+and will accept any valid rule in the following formats:
+* toml
+* json
+* yaml (yup)
+* ndjson (as long as it contains only a single rule and has the extension `.ndjson` or `.jsonl`)
+
+#### `import-rules`
+
+The primary advantage of using this command is the ability to import multiple rules at once. Multiple rule paths can be
+specified explicitly with unlimited arguments, recursively within a directory using `-d/--directory`[*](#i18n-note), or 
+a combination of both.
+
+In addition to the formats mentioned using `create-rule`, this will also accept an `.ndjson`/`jsonl` file 
+containing multiple rules (as would be the case with a bulk export).
+
+This will also strip additional fields and prompt for missing required fields.
+
+<a id="i18n-note">\* Note</a>: This will attempt to parse ALL files recursively within a specified directory.
+
+
+## Uploading rules to Kibana
```

Comment:
I partially regret adding this command for the testing/support surface area it introduced. I think we should add a note describing what the designed use case is: testing a rule before sharing back to the repo. It's just not complete and mature enough yet for me to encourage its use.

Reply:
Agree but it is there and needs explanation. I like the idea of the note - will add